internal/bminventory/inventory.go

```diff
@@ -2321,13 +2321,20 @@ func (b *bareMetalInventory) updateClusterInternal(ctx context.Context, v1Params
 
 	cluster.HostNetworks = b.calculateHostNetworks(log, cluster)
 	for _, host := range cluster.Hosts {
-		if err := b.customizeHost(host); err != nil {
+		if err = b.customizeHost(host); err != nil {
 			return nil, common.NewApiError(http.StatusInternalServerError, err)
 		}
 		// Clear this field as it is not needed to be sent via API
 		host.FreeAddresses = ""
 	}
 
+	if imageInfo := b.getImageInfo(cluster.ID); imageInfo != nil {
```

Comment:
you need to use infra env id

Reply:
ClusterID and InfraEnvID are the same here